### PR TITLE
82 Bug High Speed

### LINKED ISSUE #82: On high speed, Player doesn't collect all coins
When the UPT is higher than 1, the coins get moved from just before the player to just behind the player. Because this move is instant each tick, the collision fails to trigger.

### PR DESCRIPTION
**Pull request by mpsijm**
Will close #82 

Fixed collision while moving entities. The moveTrack method now calls a moveEntity method which makes sure that the entity calls the collide methods while moving.

<bug

### CODE REVIEW

File: src/main/java/nl/tudelft/ti2206/group9/level/Track.java [hunk 1]
```diff
@@ -78,14 +78,32 @@ public final synchronized void moveTrack(final double dist) {
...
+					moveEntity(entity, -dist);
 				}
 			}
 		}
 	}
 
 	/**
+	 * Make sure the collisions are checked over the interval that the entities
+	 * are moved.
+	 * @param first Entity that the collision is checked for.
```
Comment (thervh70) [inline src/main/java/nl/tudelft/ti2206/group9/level/Track.java]: This parameter has been removed from the method. Please refactor.

< documentation

**Comment by mpsijm** [comment]
JavaDoc has been fixed. Please keep reviewing!

<documentation,review

**Comment by MathiasMeuleman** [comment]
Looks fine!

I reviewed :)

<review

**Comment by mdingjan** [comment]
Nice bug fix :) Ready for merge.

<review

**Comment by thervh70** [comment]
Good you fixed the javadoc. Will merge

< review

**Pull request merged**